doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,236 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
+        });
+
+
+    .. note:: Note that even if the promises are created already resolved or rejected, the `then` or `catch` handlers will still be called asynchronously.
+
+
+- based on an already asynchronous code
+    Suppose that in a function you must do a rpc, and when it is completed set the result on this.
+    The `this._rpc` is a function that returns a `Promise`.
+
+    .. code-block:: javascript
+
+        function callRpc() {
+            var self = this;
+            return this._rpc(...).then(function (result) {
+                self.myValueFromRpc = result;
+            });
+        }
+
+- for callback based function
+    Suppose that you were using a function `this.close` that takes as parameter a callback that is called when the closing is finished.
+    Now suppose that you are doing that in a method that must send a promise that is resolved when the closing is finished.
+
+    .. code-block:: javascript
+        :linenos:
+
+        function waitForClose() {
+            var self = this;
+            return new Promise (function(resolve, reject) {
+                self.close(resolve);
+            });
+        }
+
+    * line 2: we save the `this` into a variable so that in an inner function, we can access the scope of our component
+    * line 3: we create and return a new promise. The constructor of a promise takes a function as parameter. This function itself has 2 parameters that we called here `resolve` and `reject`
+        - `resolve` is a function that, when called, puts the promise in the resolved state.
+        - `reject` is a function that, when called, puts the promise in the rejected state. We do not use reject here and it can be omitted.
+    * line 4: we are calling the function close on our object. It takes a function as parameter (the callback) and it happens that resolve is already a function, so we can pass it directly. To be clearer, we could have written:
+
+    .. code-block:: javascript
+
+        return new Promise (function (resolve) {
+            self.close(function () {
+                resolve();
+            });
+        });
+
+
+
+- creating a promise generator (calling one promise after the other *in sequence* and waiting for the last one)
+    Suppose that you need to loop over an array, do an operation *in sequence* and resolve a promise when the last operation is done.
+
+    .. code-block:: javascript
+
+        function doStuffOnArray(arr) {
+            var done = Promise.resolve();
+            arr.forEach(function (item) {
+                done = done.then(function () {
+                    return item.doSomethingAsynchronous();
+                });
+            });
+            return done;
+        }
+
+    This way, the promise you return is effectively the last promise.
+- creating a promise, then resolving it outside the scope of its definition (anti-pattern)
+    .. note:: we do not recommend using this, but sometimes it is useful
```

Comment:
I think the warning is not strong enough. I expected something more like "This should only be used as a last-resort when you can't figure out the proper way to do it otherwise."